ros_cross_compile/dependencies.py

```diff
@@ -95,3 +97,22 @@ def assert_install_rosdep_script_exists(
             'Rosdep installation script has never been created, you need to run this without '
             'skipping rosdep collection at least once.')
     return True
+
+
+class DependenciesStage(PipelineStage):
+    """Wrapper class for running the dependencies stage of the pipeline."""
+
+    def __init__(self):
+        self.name = gather_rosdeps.__name__
+
+    def __call__(self, platform: Platform, docker_client: DockerClient, ros_workspace_dir: Path,
+                 customizations: PipelineStageConfigOptions):
+        if not customizations.skip_rosdep_collection:
```

Comment:
This looks great - could you add a comment note here along the lines of

```
# NOTE: stage skipping will be handled more generically in the future, but to maintain the original API for now we handle this specific case internally to the stage
```

Reply:
consider aligning type names with instance names:

```
pipeline_stage_config_options: PipelineStageConfigOptions
```